src/ui/public/timefilter/timefilter.js

```diff
@@ -20,142 +20,155 @@
 import _ from 'lodash';
 import moment from 'moment';
 import { calculateBounds, getTime } from './get_time';
-import '../state_management/global_state';
-import '../config';
-import { EventsProvider } from '../events';
+import { parseQueryString } from 'ui/timefilter/lib/parse_querystring';
 import { diffTimeFactory } from './lib/diff_time';
 import { diffIntervalFactory } from './lib/diff_interval';
+import { SimpleEmitter } from 'ui/utils/simple_emitter';
 import uiRoutes from '../routes';
-import { uiModules } from '../modules';
-import { createLegacyClass } from '../utils/legacy_class';
+import chrome from 'ui/chrome';
+
+class Timefilter extends SimpleEmitter {
+  constructor() {
+    super();
+    const self = this;
+    this.diffTime = diffTimeFactory(self);
+    this.diffInterval = diffIntervalFactory(self);
+    this.isTimeRangeSelectorEnabled = false;
+    this.isAutoRefreshSelectorEnabled = false;
+    this._time = chrome.getUiSettingsClient().get('timepicker:timeDefaults');
+    this._refreshInterval = chrome.getUiSettingsClient().get('timepicker:refreshIntervalDefaults');
+  }
+
+  getTime = () => {
+    return this._time;
+  }
+
+  /**
+   * Updates timefilter time.
+   * @param {Object} time
+   * @param {string|moment} from
+   * @param {string|moment} to
+   * @param {string} mode (quick | relative | absolute)
+   */
+  setTime = (time) => {
+    this._time = Object.assign(this._time, time);
+    this.diffTime();
+  }
+
+  getRefreshInterval = () => {
+    return this._refreshInterval;
+  }
+
+  /**
+   * Set timefilter refresh interval.
+   * @param {Object} refreshInterval
+   * @param {number} value
+   * @param {boolean} pause
+   */
+  setRefreshInterval = (refreshInterval) => {
+    this._refreshInterval = Object.assign(this._refreshInterval, refreshInterval);
```

Comment:
Do you use Object.assign instead of just `this._refreshInterval = refreshInterval` because you expect refreshInterval might sometimes not be partially composed?  Or just preserving original logic?  It'll be great when we can typescript everything so this becomes obvious. :)

Reply:
I used Object.assign because of the provided value may be partially composed